link.x.in

```diff
@@ -115,6 +115,20 @@ SECTIONS
   /* LMA of .data */
   __sidata = LOADADDR(.data);
 
+  .ramfunc : ALIGN(4)
+  {
+    *(.ramfunc.*);
+
+    . = ALIGN(4); /* 4-byte align the end (VMA) of this section */
+  } > RAM AT > FLASH
```

Comment:
Do we want it at RAM? It is common to put the `ramfunc` in CCM (core coupled memory) RAM to not contend with the data RAM or vice versa.
Can we have make this user specifiable?

Reply:
> Can we have make this user specifiable?

In general, we don't have good support for multiple memory regions -- we hardcode most things to Flash and RAM (e.g. we don't have a way to say put this `static mut` in CCRAM and that one in RAM).

By user specifiable do you mean that (a) the user should be able to (a) say pull *all* `#[ramfunc]` functions into CCRAM or put all of them in RAM; or (b) that the user should be specify that this one function goes in CCRAM and that one goes in RAM?

Reply:
Preferable for each one, but if we can make all settable that would be a good improvement.

Reply:
Would be nice if we could do `#[ramfunc(section="REGION_CCRAM")]` where user would add the necessary definition or alias in memory.x. This would allow fine grained control in future I suppose?